runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -322,8 +322,174 @@ J9::ARM64::TreeEvaluator::DIVCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
       snippet->gcMap().setGCRegisterMask(0xffffffff);
       }
 
-   cg->evaluate(node->getFirstChild());
+TR::Register *
+J9::ARM64::TreeEvaluator::ArrayCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Register *obj1Reg = cg->evaluate(node->getFirstChild());
+   TR::Register *obj2Reg = cg->evaluate(node->getSecondChild());
+   TR::Register *tmp1Reg = cg->allocateRegister();
+   TR::Register *tmp2Reg = cg->allocateRegister();
+   TR::LabelSymbol *doneLabel = generateLabelSymbol(cg);
+   TR::RegisterDependencyConditions *deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(2, 2, cg->trMemory());
+   TR::Compilation* comp = cg->comp();
+   TR::LabelSymbol *snippetLabel;
+   TR::Instruction *gcPoint;
+   TR::Snippet *snippet;
+
+   TR::addDependency(deps, obj1Reg, TR::RealRegister::NoReg, TR_GPR, cg);
+   TR::addDependency(deps, obj2Reg, TR::RealRegister::NoReg, TR_GPR, cg);
+   TR::addDependency(deps, tmp1Reg, TR::RealRegister::NoReg, TR_GPR, cg);
+   TR::addDependency(deps, tmp2Reg, TR::RealRegister::NoReg, TR_GPR, cg);
+
+   // Same array, we are done.
+   generateCompareInstruction(cg, node, obj1Reg, obj2Reg);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, doneLabel, TR::CC_EQ);
+
+   // If we know nothing about either object, test object1 first. It has to be an array.
+   if ((!node->isArrayChkPrimitiveArray1())
+         && (!node->isArrayChkReferenceArray1())
+         && (!node->isArrayChkPrimitiveArray2())
+         && (!node->isArrayChkReferenceArray2()))
+      {
+#ifdef OMR_GC_COMPRESSED_POINTERS
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrw, node, tmp1Reg,
+            new (cg->trHeapMemory()) TR::MemoryReference(obj1Reg, (int32_t) TR::Compiler->om.offsetOfObjectVftField(), cg));
+#else
+      generateTrg1MemInstruction(cg, isLong ? TR::InstOpCode::ldrx : TR::InstOpCode::ldrw, node, tmp1Reg,
```

Comment:
`isLong` is declared nowhere.

Reply:
You don't need to switch the `ldr` instruction based on `TR::Compiler->target.is64Bit()` in the non-compressed pointer case.  There is no 32-bit target environment for AArch64.